src/bin/pg_autoctl/monitor.c

```diff
@@ -2554,8 +2600,19 @@ monitor_start_maintenance(Monitor *monitor, int nodeId)
 								   paramCount, paramTypes, paramValues,
 								   &context, &parseSingleValueResult))
 	{
+		if (monitor_retryable_error(context.sqlstate))
+		{
+			log_warn("Failed to start_maintenance of node %d on the monitor, "
+					 "retrying in %ds.",
+					 nodeId, PG_AUTOCTL_KEEPER_SLEEP_TIME);
+
+			sleep(PG_AUTOCTL_KEEPER_SLEEP_TIME);
```

Comment:
Can we use our exponential random backoff mechanism to implement this sleep?

Reply:
Not easily. Our dynamic retry policy depends on a state that is attached to the PGSQL object, and used privately in the context of re-connecting. The policy we set on a monitor PGSQL object is meant to defer retries to another part of the code, and so we enforce `pgsql_set_main_loop_retry_policy` on the monitor's connection, which means no retry by default.

We could change the level of abstraction where our function `pgsql_compute_connection_retry_sleep_time` works and then handle that context in the monitor's function where we implement some retry, or maybe on their callers. I can have a try at that.

Reply:
Done now, including with the registration monitor API where we already had a crude retry mechanism.